airbyte-integrations/connectors/destination-jdbc/src/main/java/io/airbyte/integrations/destination/jdbc/AbstractJdbcDestination.java

```diff
@@ -24,44 +24,39 @@
 
 package io.airbyte.integrations.destination.jdbc;
 
-import static org.jooq.impl.DSL.currentSchema;
-
 import com.fasterxml.jackson.databind.JsonNode;
 import io.airbyte.commons.json.Jsons;
-import io.airbyte.commons.lang.CloseableQueue;
 import io.airbyte.commons.resources.MoreResources;
-import io.airbyte.db.Database;
 import io.airbyte.db.Databases;
+import io.airbyte.db.jdbc.JdbcDatabase;
+import io.airbyte.db.jdbc.JdbcUtils;
 import io.airbyte.integrations.base.Destination;
 import io.airbyte.integrations.base.DestinationConsumer;
-import io.airbyte.integrations.destination.DestinationConsumerFactory;
 import io.airbyte.integrations.destination.NamingConventionTransformer;
-import io.airbyte.integrations.destination.SqlDestinationOperations;
 import io.airbyte.protocol.models.AirbyteConnectionStatus;
 import io.airbyte.protocol.models.AirbyteConnectionStatus.Status;
 import io.airbyte.protocol.models.AirbyteMessage;
 import io.airbyte.protocol.models.ConfiguredAirbyteCatalog;
 import io.airbyte.protocol.models.ConnectorSpecification;
 import java.io.IOException;
-import java.sql.SQLException;
-import org.jooq.DSLContext;
-import org.jooq.SQLDialect;
+import java.util.function.Function;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public abstract class AbstractJdbcDestination implements Destination {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(AbstractJdbcDestination.class);
-  protected static final String COLUMN_NAME = "data";
 
   private final String driverClass;
-  private final SQLDialect dialect;
   private final NamingConventionTransformer namingResolver;
+  private final Function<JdbcDatabase, SqlOperations> sqlOperationsFactory;
 
-  public AbstractJdbcDestination(final String driverClass, final SQLDialect dialect, final NamingConventionTransformer namingResolver) {
+  public AbstractJdbcDestination(final String driverClass,
+                                 final NamingConventionTransformer namingResolver,
+                                 final Function<JdbcDatabase, SqlOperations> sqlOperationsFactory) {
```

Comment:
It is really weird to pass a creator instead of the an actual operation.

I see that the SQLOperation class uses the database as a member. I think we should pass it as an argument in method that require it this way you don't need the database object to create the sqloperation.